### PR TITLE
Fix barcode scanner issues + scroll edit text

### PR DESCRIPTION
**Pull request by ignaciocervinoGlob**
### Description
Fixing barcode scanner bugs and adjust text field when editing.

### Tasks
This PR solves 2 tickets:
#### Scroll to see which field I'm editing:
- [x] Adjust in add asset form activity

#### Fix Existing flow of Existing Asset on add activity:
- [x] When user scan a barcode in add option and already exist, go to the update form
- [x] Camera stops scanning when going back or dismissing alerts
- [x] In search asset, if user press search manually button, camera must stop searching for barcode
- [x] In add asset if user go back to camera, the scanner is disabled
- [x] @globant in update
- [x] Check without internet cases
- [x] When there is no internet in camera, user can still scan and loader shows, same if shutdown internet during the activity
- [x] When there is no internet in asset list and user selects an asset, add from shows network alert with a loader runing
- [x] Search Alert dismiss when touch outside it
- [x] Fix UT 

#### Scroll to see which field I'm editing
https://user-images.githubusercontent.com/90403672/155404324-f58b3366-7d82-44c4-a96d-b3223a1fc574.mp4

##### When user scan a barcode in add option and already exist, go to the update form
https://user-images.githubusercontent.com/90403672/155405969-c8fd9014-f046-4bcc-931c-7714f879699c.mp4

#### Search asset flow
https://user-images.githubusercontent.com/90403672/155406148-0ed8a659-73dd-45d1-a570-2ce5140c4e8a.mp4

#### User can't scan when dialog shows
https://user-images.githubusercontent.com/90403672/155448565-7110b95f-0b85-4e80-b56d-488a5e79112f.mp4

#### No internet when scanning
https://user-images.githubusercontent.com/90403672/155448675-a17b0462-3530-4ffd-bb17-5ff3d146b786.mp4

##### Update asset from asset list → no internet
https://user-images.githubusercontent.com/90403672/155448786-0f7945c8-e928-4c24-91ee-2b9fc2de92f1.mp4

### Coverage
image

### CODE REVIEW

File: app/src/main/java/com/mobiletandil/stockapp/viewModel/AssetFormViewModel.kt [hunk 1]
```diff
@@ -223,7 +223,7 @@ class AssetFormViewModel(
         ASSET_BY_BARCODE_DO_NOT_EXIST,
         GO_TO_MAIN_ACTIVITY,
         EMPTY_MAIL,
-        SET_ASSET_UPDATE_FORM
+        SET_ASSET_UPDATE_FORM,
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/viewModel/AssetFormViewModel.kt:226]: remove this extra comma

File: app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt [hunk 2]
```diff
@@ -108,7 +115,7 @@ class ScannerActivity : AppCompatActivity() {
                         getString(R.string.asset_form_dialog_title),
                         getString(R.string.asset_form_change_to_update_mode_dialog),
                         getString(R.string.asset_form_dialog_button_ok_title),
-                        { startActivity(eventData.value.let { AssetInformationListActivity.newIntent(this, BARCODE_FILTER, it) }) },
+                        { startActivity(AssetFormActivity.newIntent(this, eventData.value, true, eventData.asset)) },
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt:118]: check this please, remember that from here you should call the viewModel. Check it in all this class

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mobiletandil/stockapp/viewModel/AssetFormViewModel.kt (hunk 1 above)
Comment (ignaciocervinoGlob) [reply app/src/main/java/com/mobiletandil/stockapp/viewModel/AssetFormViewModel.kt]: done

File: app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt (hunk 2 above)
Comment (ignaciocervinoGlob) [reply app/src/main/java/com/mobiletandil/stockapp/activity/ScannerActivity.kt]: done

File: app/src/main/java/com/mobiletandil/stockapp/viewModel/SearchPickerViewModel.kt [hunk 3]
```diff
@@ -107,7 +107,7 @@ class SearchPickerViewModel(private val getLocationsUseCase: GetLocationsUseCase
         }
     }
 
-    private fun getLocations() = viewModelScope.launch {
+    fun getLocations() = viewModelScope.launch {
```
Comment (SartiMau) [inline app/src/main/java/com/mobiletandil/stockapp/viewModel/SearchPickerViewModel.kt:110]: is this change okay? I remember that we fix this with @Georgina-Fernandez

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/mobiletandil/stockapp/viewModel/SearchPickerViewModel.kt (hunk 3 above)
Comment (ignaciocervinoGlob) [reply app/src/main/java/com/mobiletandil/stockapp/viewModel/SearchPickerViewModel.kt:110]: It's for testing purpose, if it's private I don't find the way to test it

**Review by SartiMau — APPROVED** [review]

**Review by GARAF004 — APPROVED** [review]

**Review by MarcosCGlobant — APPROVED** [review]

**Pull request merged**